doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,435 @@
+
+* str - measured in bytes, the length of the string must not be greater than the key_str modulus length in bytes - 11
+
+* key_str - a key (either private or public) in the PEM format
+
+.. _asym-sign:
+
+asymmetric_sign(*algorithm, digest_str, priv_key_str, digest_type*)
+------------------------------------------------------------------------
+
+Signs a digest string using a private key string.
+
+.. rubric:: Returns
+
+A signature is a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA' or 'DSA'.
+
+* digest_str - the digest binary string that is signed. The string can be generated by invoking :ref:`create-digest<create-digest>`.
+
+* priv_key_str - the private key used to sign the digest string. The key must be in the PEM format.
+
+* digest_type - the supported values are listed in the :ref:`create-digest<create-digest>` digest type table.
+
+
+.. _asym-verify:
+
+asymmetric_verify(*algorithm, digest_str, sig_str, pub_key_str, digest_type*)
+--------------------------------------------------------------------------------
+
+Verifies whether the signature string matches the digest string.
+
+.. rubric:: Returns
+
+A ``1`` (success) or a ``0`` (failure).
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA' or 'DSA'.
+
+* digest_str - the digest binary string and this string can be generated by invoking :ref:`create-digest<create-digest>`.
+
+* sig_str - the signature binary string. This string can be generated by invoking :ref:`asymmetric_sign<asym-sign>`.
+
+* pub_key_str - the signer's public key string. This string must correspond to the private key passed to :ref:`asymmetric_sign<asym-sign>` to generate the signature string. The string must be in the PEM format.
+
+* digest_type - the supported values are listed in the :ref:`create-digest<create-digest>` digest type table.
+
+.. _create-priv-key:
+
+create_asymmetric_priv_key(*algorithm, (key_len | dh_parameters)*)
+--------------------------------------------------------------------------------
+
+Generates a private key using the given algorithm and key length for RSA or DSA or Diffie-Hellman parameters (for DH).
+
+.. rubric:: Returns
+
+The key as a string in PEM format.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA','DSA', or 'DH'.
+
+* key_len - the supported key length values are the following:
+
+  - RSA - the minimum length is 1,024. The maximum length is 16,384. The default value is 16,384.
+
+  - DSA - the minimum length is 1,024. The maximum length is 9,984. The default value is 9,984.
+
+    .. note:: The key length limits are defined by OpenSSL. To change the maximum key length, use either :ref:`rsa-threshold` or :ref:`dsa-threshold`.
+
+* dh_parameters - a Diffie-Hellman (DH) parameter. You can create DH parameters with the :ref:`create-dh-parameter<create-dh-param>`.
```

Comment:
Replace create-dh-parameter => create_dh_parameter

Reply:
all dashes changed to underscores